oak_runtime/src/node/http/tests.rs

```diff
@@ -151,17 +214,59 @@ async fn test_https_server_does_not_terminate_after_a_bad_request() {
     let client_with_invalid_tls = create_client(GCP_CA);
 
     // Send a valid request, making sure that the server is started
-    let resp = send_request(client_with_valid_tls.clone(), "https://localhost:2527").await;
+    let resp = send_request(
+        client_with_valid_tls.clone(),
+        "https://localhost:2527",
+        Label::public_untrusted(),
+    )
+    .await;
     assert!(resp.is_ok());
 
     // Send an HTTPS request with invalid certificate, and check that the server responds with error
-    let resp = send_request(client_with_invalid_tls, "https://localhost:2527").await;
+    let resp = send_request(
+        client_with_invalid_tls,
+        "https://localhost:2527",
+        Label::public_untrusted(),
+    )
+    .await;
     assert!(resp.is_err());
 
     // Send another valid request, and check that the server is alive and responsive
     // let client_with_valid_tls = create_client(LOCAL_CA);
-    let resp = send_request(client_with_valid_tls, "https://localhost:2527").await;
+    let resp = send_request(
+        client_with_valid_tls,
+        "https://localhost:2527",
+        Label::public_untrusted(),
+    )
+    .await;
+    assert!(resp.is_ok());
+
+    // Stop the runtime and the servers
+    http_server_tester.cleanup();
+}
+
+#[tokio::test]
+async fn test_https_server_can_serve_https_requests_with_non_empty_request_label() {
+    init_logger();
+
+    // Start a runtime with an HTTP server node, and a thread simulating an Oak node to respond to
+    // HTTP requests.
+    let mut http_server_tester = HttpServerTester::new(2528);
+    let client_with_valid_tls = create_client(LOCAL_CA);
+
+    // Send an HTTPS request, and check that response has StatusCode::OK
+    let resp = send_request(
+        client_with_valid_tls,
+        "https://localhost:2528",
+        confidentiality_label(tls_endpoint_tag("localhost")),
+    )
```

Comment:
As I mentioned in the other comment, this only works because the label is exactly the one that is used above, but it would be nice to show that multiple labels are declassified. WDYT?

Reply:
I am not sure that this should be the purpose of the HTTP tests. But I agree that the label should be retrieved from the request. I'll fix that.